src/ui.rs

```diff
@@ -116,6 +167,20 @@ impl Ui {
         }
     }
 
+    /// Switches the output to use the pager, if allowed.
+    pub fn request_pager(&mut self) {
+        if !matches!(self.paginate, PaginationChoice::Auto) {
+            return;
+        }
+        if matches!(self.output, UiOutput::Paged { .. }) {
+            return;
+        }
+        if !atty::is(Stream::Stdout) {
```

Comment:
It looks like it should never make a difference, but what do you think about checking for the Stdout in `self.output`? Combined with the case above, that might look like this:

```
match &self.output {
  UiOutput::Paged { .. } => {
    return;
   }
  UiOutput::Terminal { stdout, ..} => {
    if !atty::is(stdout) {
      return;
    }
  }
}
self.output = UiOutput::new_paged_else_terminal(&self.settings);
```

Actually, maybe we should invert that to be
```
match &self.output {
  UiOutput::Paged = {}
  UiOutput::Terminal { stdout, ..} => {
    if atty::is(stdout) {
      self.output = UiOutput::new_paged_else_terminal(&self.settings);
    }
  }
}
```
You decide.

Reply:
I think `atty::is(Stream::Stdout)` is a bit clearer because it's closer to "Is the stdout connected to a tty?" as opposed to `atty::is(stdout)`, which technically is always the `stdout`, but with an extra bit of indirection through `UiOutput::Terminal`.

Also, I think `atty::is` accepts an internal `Stream` enum with variants corresponding to the standard fd-s, not the structs from `std::io`

Reply:
To clarify, the reason I suggested `atty::is(stdout)` was that if `stdout` was not actually connected to stdout, then it would probably be better to check whatever it was connected to. But since that's not how `atty::is()` works, that's clearly not an option :)

I would still prefer to restructure the conditions using `match` instead of `matches!()`

Reply:
I don't think the `Stdout` type can represent anything but the hardcoded file descriptor, anyway.

Reply:
The match was done. `atty` got replaced by `crossterm`